Tests/NIOTransportServicesTests/NIOTSConnectionChannelTests.swift

```diff
@@ -865,5 +865,61 @@ class NIOTSConnectionChannelTests: XCTestCase {
             .wait()
         XCTAssertNoThrow(try connection.close().wait())
     }
+
+    func testErrorIsForwardedFromFailedConnectionState() throws {
+        final class ForwardErrorHandler: ChannelDuplexHandler {
+            typealias OutboundIn = ByteBuffer
+            typealias InboundIn = ByteBuffer
+            
+            private let testCompletePromise: EventLoopPromise<Error>
+            let listenerChannel: Channel
+            
+            init(testCompletePromise: EventLoopPromise<Error>, listenerChannel: Channel) {
+                self.testCompletePromise = testCompletePromise
+                self.listenerChannel = listenerChannel
+            }
+            func channelActive(context: ChannelHandlerContext) {
+                let listenerErrorPromise = self.listenerChannel.eventLoop.next().makePromise(of: Error.self)
+                listenerChannel
+                    .close()
+                    .whenSuccess { _ in
+                        context.channel.write(ByteBuffer(data: Data()))
+                            .whenFailure({ error in
+                                listenerErrorPromise.succeed(error)
+                            })
+                    }
+            }
+            
+            func errorCaught(context: ChannelHandlerContext, error: Error) {
+                let error = error as? ChannelError
+                XCTAssertNotEqual(error, ChannelError.eof)
+                XCTAssertEqual(error, ChannelError.ioOnClosedChannel)
+                XCTAssertNotNil(error)
+                testCompletePromise.succeed(error!)
+            }
+        }
+        
+        let listener = try NIOTSListenerBootstrap(group: self.group)
+            .childChannelInitializer { channel in
+                return channel.eventLoop.makeSucceededVoidFuture()
+            }
+            .bind(host: "127.0.0.1", port: 8080)
```

Comment:
Can we `bind` `0` instead? This can cause awkward conflicts if this port is in use somewhere else.